File: ClientApp/src/app/models/request-group.ts
import { RequestStatus } from "./request-status.enum";
import { Request } from "./request";
import { User } from "./user";
import { RequestType } from "./request-type.enum";

export interface RequestGroup {
  id: string;
  dateRequested: Date;
  userId: number;
  user?: User;
  managerId: number;
  status: RequestStatus;
  requests: Request[];
  type: RequestType;
}

File: Models/RequestGroup.cs
using System;
using System.Collections.Generic;

public class RequestGroup {
  public string id {get;set;}
  public DateTime dateRequested {get;set;}
  public int userId {get;set;}
  public User user { get; set; }
  public int managerId {get;set;}
  public RequestStatus status {get;set;}
  public List<Request> requests {get;set;}
}
